### PR TITLE
feat: FO-1749 | Add support for react elements in flash message

### PR DESCRIPTION
**Pull request by akshayjain04**
_Linked issue, attach here._
None

_Is it a_

- [ ] Bug
- [x] Feature

## Requirements

- [x] Read the [contribution guidelines](./.github/CONTRIBUTING.md).
- [x] Updated docs if applicable.

## Rationale

_Why is this PR necessary?_

We require to have flash messages to support react elements like adding links to them etc. To achieve this, this PR aims to update typescript type of `message` to support react elements.

### CODE REVIEW

**Comment by radhika0406** [comment]
> _Linked issue, attach here._
> 
> _Is it a_
> 
> * [ ]  Bug
> * [x]  Feature
> 
> ## Requirements
> * [ ]  Read the [contribution guidelines](./.github/CONTRIBUTING.md).
> 
> * [ ]  Updated docs if applicable.
> 
> ## Rationale
> _Why is this PR necessary?_
> 
> Updated `message` type to support react elements in flash messages.

@akshayjain04 is it possible to copy the PR description template we follow in here & update description accordingly?

File: src/actions.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
+import React from 'react';
 import { ADD_MESSAGE, CLEAR_MESSAGE, CLEAR_ALL_MESSAGES } from './constants';
 import { Message } from './factory';
 
-export type AddMessageParams = Partial<Message> & { message: string };
+export type AddMessageParams = Partial<Message> & { message: string | React.ReactNode };
```
Comment (peak-jiteshahuja) [inline src/actions.ts:5]:
```suggestion
export type AddMessageParams = Partial<Message> & Pick< Message , 'message'>;
```
Nit.

File: yarn.lock [hunk 2]
```diff
@@ -64,6 +64,25 @@
   resolved "https://registry.yarnpkg.com/@types/parse-json/-/parse-json-4.0.0.tgz#2f8bb441434d163b35fb8ffdccd7138927ffb8c0"
   integrity sha512-//oorEZjL6sbPcKUaCdIGlIUeH26mgzimjBB77G6XRgnDl/L5wOnpyBGRe/Mmf5CVW3PwEBE1NjiMZ/ssFh4wA==
 
+"@types/prop-types@*":
```
Comment (peak-jiteshahuja) [inline yarn.lock:67]: Hey @akshayjain04 - Do we need proptypes ? 🤔 Or was it automatically added ?

**Review by peak-jiteshahuja — APPROVED** [review]

File: yarn.lock (hunk 2 above)
Comment (akshayjain04) [reply yarn.lock:67]: Hey @peak-jiteshahuja, it was automatically added

File: src/actions.ts (hunk 1 above)
Comment (akshayjain04) [reply src/actions.ts]: Updated

**Review by saurabh0402 — DISMISSED** [review]

**Review by peak-jiteshahuja — APPROVED** [review]

**Review by peak-jiteshahuja — APPROVED** [review]
Hey @akshayjain04 - Can we please update readme to update release process.

**Review by peak-jiteshahuja — DISMISSED** [review]
Hey @akshayjain04 - Can we pl update PR description ?

**Review by radhika0406 — APPROVED** [review]

**Review by saurabh0402 — APPROVED** [review]

**Review by radhika0406 — APPROVED** [review]

**Review by azhar22k — APPROVED** [review]

**Review by pearcem0 — APPROVED** [review]

**Pull request merged**